### PR TITLE
tgui v3

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
https://github.com/tgstation/tgstation/pull/50497

## Changelog
:cl:
code: tgui updated
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/APC.js [hunk 1]
```diff
@@ -1,82 +1,50 @@
...
+import { useBackend } from '../backend';
+import { Box, Button, LabeledList, NoticeBox, ProgressBar, Section } from '../components';
+import { Window } from '../layouts';
 import { InterfaceLockNoticeBox } from './common/InterfaceLockNoticeBox';
 
-export const Apc = props => {
-  const { state } = props;
-  const { config, data } = state;
-  const { ref } = config;
+export const Apc = (props, context) => {
```
Comment (psykzz) [inline tgui/packages/tgui/interfaces/APC.js:7]:
```suggestion
export const APC = (props, context) => {
```

Comment (spookydonut) [reply tgui/packages/tgui/interfaces/APC.js:7]: I need to rename the file actually

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -666,7 +666,7 @@
 	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 
 	if(!ui)
-		ui = new(user, src, ui_key, "apc", name, ui_x, ui_y, master_ui, state)
+		ui = new(user, src, ui_key, "APC", name, ui_x, ui_y, master_ui, state)
```
Comment (spookydonut) [inline code/modules/power/apc.dm:669]:
```suggestion
		ui = new(user, src, ui_key, "Apc", name, ui_x, ui_y, master_ui, state)
```

File: tgui/packages/tgui/interfaces/APC.js (hunk 1 above)
Comment (psykzz) [reply tgui/packages/tgui/interfaces/APC.js]: either or would've wokred

**Comment by David-Stormwell** [comment]
Stuff like medic vendors and other vendors which use a point system most likely don't work, as in you cannot buy anything because it's simply disappeared.

**Comment by spookydonut** [comment]
I tested all uis and they worked for me can you be more specific

**Comment by psykzz** [comment]
This seems to just be something with the sulaco, and their vendors there.
https://github.com/tgstation/TerraGov-Marine-Corps/pull/3891#issuecomment-619252653

**Comment by psykzz** [comment]
unless you have examples on other maps @David-Stormwell

**Pull request merged**